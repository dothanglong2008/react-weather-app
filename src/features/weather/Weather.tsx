import { useAppDispatch, useAppSelector } from "../../routes/app/hooks";
import styles from "./Weather.module.scss";
import { selectWeather } from "./weatherSlice";

interface WeatherProps {}

function Weather() {
  const weather = useAppSelector(selectWeather);
  const dispatch = useAppDispatch();

  return (
    <div className={styles.Weather}>
      <header className={styles.flex}>
        <div>
          <h1>React App Weather</h1>
        </div>
        <div>Search Bar</div>
      </header>
      <main className={styles.flex}></main>
    </div>
  );
}

export default Weather;
