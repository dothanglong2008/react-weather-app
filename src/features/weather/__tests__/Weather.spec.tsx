import Weather from "../Weather";
import renderer from "react-test-renderer";
import { cleanup, render, screen } from "@testing-library/react";

type Container = HTMLDivElement | null;

let container: Container = null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(cleanup);

it("renders Hello text", () => {
  render(<Weather />);
  const text = screen.getByText(/Hello/i);
  expect(text).toBeInTheDocument();
});

it("match snapshot with empty text", () => {
  const component = renderer.create(<Weather />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
