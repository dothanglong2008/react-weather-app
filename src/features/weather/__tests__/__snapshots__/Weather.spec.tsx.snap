// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Weather component renders Weather correctly 1`] = `
Array [
  <header
    className="header"
  >
    <h1
      className="heading"
    >
      <svg
        baseProfile="tiny"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        version="1.2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M14.5 3l-1 3-1-3c-.184-.553.114-1.149.666-1.333.553-.185 1.15.114 1.334.666.075.226.07.458 0 .667zM19.864 6.05l-2.829 1.415 1.415-2.829c.261-.521.894-.731 1.414-.472.521.261.731.894.472 1.415-.107.212-.274.372-.472.471zM21.5 12l-3-1 3-1c.553-.185 1.149.114 1.334.667.184.552-.115 1.148-.668 1.333-.225.075-.457.069-.666 0zM8.55 4.636l1.415 2.829-2.829-1.415c-.521-.261-.732-.894-.472-1.414.261-.521.895-.731 1.414-.472.213.107.373.274.472.472zM17.776 12.342c.139-.424.224-.871.224-1.342 0-2.481-2.019-4.5-4.5-4.5-1.34 0-2.537.594-3.357 1.528l-.143-.028c-1.776 0-3.369.78-4.469 2.011-.24-.08-.472-.086-.697-.011-.553.185-.852.781-.668 1.333.057.167.158.299.277.411-.283.697-.443 1.458-.443 2.256l.002.126c-1.725.445-3.002 2.013-3.002 3.874 0 2.206 1.795 4 4 4h11c2.757 0 5-2.243 5-5 0-2.129-1.344-3.939-3.224-4.658zm-4.276-3.842c1.379 0 2.5 1.121 2.5 2.5 0 .366-.096.706-.238 1.019-.354.021-.72.074-1.118.188-.521-1.353-1.604-2.415-2.967-2.905.456-.49 1.102-.802 1.823-.802zm2.5 11.5h-11c-1.104 0-2-.897-2-2s.896-2 1.908-2.006l1.422.016-.248-1.202c-.055-.263-.082-.536-.082-.808 0-2.206 1.795-4 4-4l.069-.014c1.904.055 3.495 1.406 3.847 3.27l.038.186c.123.436.517.706.946.712l.289-.023c.312-.09.569-.131.811-.131 1.654 0 3 1.346 3 3s-1.346 3-3 3z"
          />
        </g>
      </svg>
      <span>
        React Weather App
      </span>
    </h1>
    <aside
      className="sidebar"
    >
      <form
        className="form"
        name="search"
        onSubmit={[Function]}
      >
        <div
          className="searchContainer"
        >
          <div
            className="inputContainer"
          >
            <input
              className="input"
              id="location"
              name="location"
              onChange={[Function]}
              placeholder="Search Location"
              type="text"
              value=""
            />
            <button
              className="suggestion"
              onClick={[Function]}
              title="Select current location"
              type="button"
            >
              <svg
                color="grey"
                fill="currentColor"
                height="2em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": "grey",
                  }
                }
                viewBox="0 0 24 24"
                width="2em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="4"
                />
                <path
                  d="M13 4.069V2h-2v2.069A8.01 8.01 0 0 0 4.069 11H2v2h2.069A8.008 8.008 0 0 0 11 19.931V22h2v-2.069A8.007 8.007 0 0 0 19.931 13H22v-2h-2.069A8.008 8.008 0 0 0 13 4.069zM12 18c-3.309 0-6-2.691-6-6s2.691-6 6-6 6 2.691 6 6-2.691 6-6 6z"
                />
              </svg>
            </button>
            <button
              aria-label="search"
              className="searchButton"
              type="submit"
            >
              <svg
                color="grey"
                fill="currentColor"
                height="1.5em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": "grey",
                  }
                }
                viewBox="0 0 512 512"
                width="1.5em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                />
              </svg>
            </button>
          </div>
        </div>
      </form>
    </aside>
  </header>,
  <main
    className="main"
  >
    <p>
      Search for a city or get current location
    </p>
  </main>,
  <footer
    className="footer"
  >
    <p>
      <span>
        API Source: 
      </span>
      <a
        href="https://www.weatherapi.com/"
      >
        weatherapi.com
      </a>
    </p>
  </footer>,
]
`;
